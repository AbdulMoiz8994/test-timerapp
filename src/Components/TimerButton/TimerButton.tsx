import React from 'react'

interface Props{
   buttonAction:() => void
   buttonValue: string
}

export const TimerButton: React.FC<Props> = ({buttonAction,buttonValue}) => {
    return (
        <div>
            <button onClick={buttonAction}>{buttonValue}</button>
        </div>
    )
}
