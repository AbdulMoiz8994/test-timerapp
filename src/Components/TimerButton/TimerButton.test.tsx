import React from 'react'
import {shallow, ShallowWrapper} from 'enzyme'
import {TimerButton} from './TimerButton'



describe('Timer Comp',() =>{
    it('Auth To Div',() =>{
        const container=shallow(<TimerButton buttonAction={jest.fn} buttonValue={""}/>)
        expect(container.find('div').length).toEqual(1)
    })

    it("Button",() =>{
        const container:ShallowWrapper=shallow(<TimerButton buttonAction={jest.fn} buttonValue={""}/>)
       expect(container.find('button').length).toBe(1)
    })
    it("Button SnapShot",() =>{
        const contianer=shallow(<TimerButton buttonAction={jest.fn} buttonValue={""}/>)
        expect(contianer).toMatchSnapshot()
    })
})

