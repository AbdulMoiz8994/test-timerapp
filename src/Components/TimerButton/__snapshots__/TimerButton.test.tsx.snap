// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer Comp Button SnapShot 1`] = `
<div>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`Timer Comp Button Text 1`] = `
<div>
  <button
    onClick={[Function]}
  />
</div>
`;
